common/src/main/scala/explore/components/AgGridHelpers.scala

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.components
+
+import scala.annotation.unused
+
+import cats._
+import cats.data.Validated._
+import cats.syntax.all._
+import cats.effect.IO
+import crystal.ViewF
+import crystal.react.implicits._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.string._
+import explore.AppCtx
+import explore._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.MonocleReact._
+import lucuma.core.util.Display
+import lucuma.core.util.Enumerated
+import lucuma.ui.forms._
+import lucuma.ui.optics._
+import monocle.Lens
+import monocle.macros.Lenses
+import org.scalajs.dom.document
+import org.scalajs.dom.ext.KeyCode
+import org.scalajs.dom.html
+import react.semanticui.elements.button.Button
+import reactST.agGridCommunity.colDefMod._
+import reactST.agGridCommunity.iCellEditorMod.ICellEditorParams
+import reactST.agGridCommunity.iCellRendererMod.ICellRendererParams
+import reactST.agGridReact.components._
+
+import scalajs.js
+
+object AgGridHelpers {
+
+  /**
+   * Returns a display only AgGridColumn Builder.
+   * Since this returns a Builder, additional column configuration methods can
+   * be chained onto the return value.
+   *
+   * @param lens A Lens to go between the Row value and the Cell value.
+   * @param formatter Function for formatting the value. This will often be the `reverseGet` of a Format or ValidFormatInput.
+   * @return An AgGridColumn Builder.
+   */
+  def viewColumn[A, B](lens: Lens[A, B])(formatter: B => String): AgGridColumn.ColDef.Builder =
+    AgGridColumn.ColDef
+      .valueGetter(getter(lens))
+      .cellRendererFramework(renderer(formatter))
+
+  /**
+   * Create a AgGridColumn Builder for a column which edits the value in a FormInputEv.
+   * Since this returns a Builder, additional column configuration methods can
+   * be chained onto the return value.
+   *
+   * @param lens Lens to go between the Row value and the Cell value.
+   * @param validFormat The ValidFormatInput for the editor.
+   * @param changeAuditor The ChangeAuditor for the editor.
+   * @param disabled Whether editing should be disabled or not.
+   * @return An AgGridColumn Builder
+   */
+  def editableViewColumn[A, B](
+    lens:          Lens[A, B],
+    validFormat:   ValidFormatInput[B] = ValidFormatInput.id,
```

Comment:
I would recommend against using `ValidFormatInput.id`. I regret using it in `FormInputEV`. It has caused me some headeaches since it forces `B = String`.

Reply:
It's caught by the compiler, though, right? Still, it's not generally what we want, so it could be surprizing. I'll just remove the default and require the passing of a ValidFormatInput.